src/braket/experimental/autoqasm/api/api.py

```diff
@@ -53,21 +54,25 @@ def function(f: Callable) -> Callable[[Any], aq_program.Program]:
         return f
 
     # Update documentation with user configuration
-    if f.__doc__ is None:
-        f.__doc__ = ""
-    f.__doc__ += f"""
+    try:
+        if f.__doc__ is None:
+            f.__doc__ = ""
+        f.__doc__ += f"""
 
 Keyword Args:
     {aq_program.ProgramOptions.NUM_QUBITS.value} (int): Configuration to set the total number of
         qubits to declare in the program.
 """
+    except Exception as e:
```

Comment:
What kind of Exception?

Reply:
AttributeError. Yes, I'll update this to be more granular. Occurs when the decorated object does not have a writable `__doc__` property. This would likely lead to another exception later (e.g., the object is not callable, etc.) but it's more informative to have that later exception thrown, rather than this one. I'll add a comment here describing this as well.